src/lib/network_pool/transaction_pool.ml

```diff
@@ -619,134 +647,10 @@ struct
               [ ( "cmd"
                 , Transaction_hash.User_command_with_valid_signature.to_yojson
                     cmd ) ] ;
-          Hashtbl.find_and_remove t.locally_generated_uncommitted cmd |> ignore
-      ) ;
+          find_and_remove_uncommitted t cmd |> ignore ) ;
       t.pool <- pool ;
       Deferred.unit
 
-    let create ~constraint_constants ~consensus_constants ~time_controller
```

Comment:
For reviewer: All I did was move this below Diff.apply and add the pipes that get written to when the locally_generated_uncommitted hash table is modified.

Reply:
Corresponding diff is

```diff
@@ -619,14 +647,17 @@ struct
               [ ( "cmd"
                 , Transaction_hash.User_command_with_valid_signature.to_yojson
                     cmd ) ] ;
-          Hashtbl.find_and_remove t.locally_generated_uncommitted cmd |> ignore
-      ) ;
+          find_and_remove_uncommitted t cmd |> ignore ) ;
       t.pool <- pool ;
       Deferred.unit

     let create ~constraint_constants ~consensus_constants ~time_controller
         ~frontier_broadcast_pipe ~config ~logger ~tf_diff_writer =
       let t =
+        let r, w =
+          Strict_pipe.create ~name:"locally-generated-changed"
+            (Buffered (`Capacity 0, `Overflow Drop_head))
+        in
         { pool=
             Indexed_pool.empty ~constraint_constants ~consensus_constants
               ~time_controller
@@ -638,6 +669,8 @@ struct
             Hashtbl.create
               ( module Transaction_hash.User_command_with_valid_signature.Stable
                        .Latest )
+        ; uncommitted_changed_r= r
+        ; uncommitted_changed_w= w
         ; config
         ; logger
         ; batcher= Batcher.create config.verifier
@@ -645,6 +678,22 @@ struct
         ; recently_seen= Lru_cache.Q.create ()
         ; best_tip_ledger= None }
       in
+      don't_wait_for
+        (let%map () =
+           match%map
+             Reader.load_bin_prot config.disk_location
+               Persisted.Stable.Latest.bin_reader_t
+           with
+           | Ok {old_locally_generated_uncommitted} ->
+               periodically_re_add_persisted_transactions t
+                 old_locally_generated_uncommitted ~frontier_broadcast_pipe
+           | Error e ->
+               [%log' debug t.logger]
+                 ~metadata:[("error", `String (Error.to_string_hum e))]
+                 !"Unable to load sent-but-unconfirmed transactions from disk \
+                   ($error)"
+         in
+         persist_on_change t) ;
       don't_wait_for
         (Broadcast_pipe.Reader.iter frontier_broadcast_pipe
            ~f:(fun frontier_opt ->
@@ -707,7 +756,7 @@ struct
                          find_remove_bool t.locally_generated_committed
                        in
                        let dropped_uncommitted =
-                         find_remove_bool t.locally_generated_uncommitted
+                         modify_uncommitted t ~f:find_remove_bool
                        in
                        (* Nothing should be in both tables. *)
                        assert (not (dropped_committed && dropped_uncommitted)) ;
```